pkg/acceptance/helpers/test_client.go

```diff
@@ -0,0 +1,29 @@
+package helpers
+
+import "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+
+type TestClient struct {
+	context *TestClientContext
+
+	Database *DatabaseClient
+}
+
+func NewTestClient(c *sdk.Client, database string, schema string, warehouse string) *TestClient {
+	context := &TestClientContext{
+		client:    c,
+		database:  database,
+		schema:    schema,
+		warehouse: warehouse,
+	}
+	return &TestClient{
+		context:  context,
+		Database: NewDatabaseClient(context),
```

Comment:
Hmm, I'm wondering that in some cases we would pass TestClient instead or additionally next to the context, right? It will be kind of circular dependency (though not causing an error), but it would be beneficial to create more complex things like applications.

Reply:
Currently, most helpers are simple, touching just one object. For complex cases we can have dedicated helper methods (in the future).